### ISSUE TITLE
Recovering old Epichrome app instances to newly created ones

### ISSUE THREAD

**Issue by ylluminate** [dmarmor/epichrome#263]
So @dmarmor we've talked about this various times before, but I wanted to recap and recapitulate the process in a dedicated ticket here in light of all the various chatter and developments that have occurred. 

I'm going to be working from a fresh new user account on my Catalina system and want to migrate my older Epichrome app configs from versions of Epichrome that haven't been used from a couple years ago to more recent versions from this past year and even from the recent'ish beta releases.

Can you please reiterate, in a stepwise bullet-point fashion, the process of taking the appropriate contents of `/Users/USERNAME/Library/Application Support/Epichrome/Apps/SOME_APP_NAME` and inserting them into a new Epichrome app instance built with a current release version?

**Comment by dmarmor** [comment]
So is your plan to create new apps rather than updating your old ones? If you're not in a super hurry to get this done, I'd recommend waiting for Epichrome 2.4, which will have app editing, as well as bulk updating of old apps, which should make the process a lot simpler.

But if you need to do it now, then the procedure should be very simple for apps that are 2.3.0b7 or later:

1. Make sure you've completely removed the old app (like with `rm` or emptying the trash after deleting it).
1. When you create the new app make sure you give the *exact* same short name (step 2 in Epichrome). Then it should have the same ID and should automatically reconnect to the same data directory.
1. If for whatever reason it doesn't end up with the same ID as the old app, then make sure the new app is not running, note the name of its data directory (which corresponds to its ID), delete that directory, and rename the old one to the same name.

For older apps, 2.2.4 and earlier, the procedure is largely the same, but before you run the new app, you must migrate the data directory. Using the example of an app with ID `SomeApp`:

1. Inside the app's data directory at `~/Library/Application Support/Epichrome/Apps/SomeApp` create a subfolder called `UserData`.
1. Move everything else in the data directory into `UserData`.

You'll get a "Welcome to your new app" message, but any state or sessions you had saved should be intact.

Let me know how it goes!

**Comment by ylluminate** [comment]
Wow - that's really nice to hear re: 2.4.  I may wait, but doubtful...

Part of the impetus here is that I'm still struggling with a previous issue I'd noted re: a very, very, very difficult to track and debilitating intermittent hang.  I've reinstalled my system with fresh system settings but the hang remained.  I created a new system account and didn't seem to have a significant hang any longer, but then I started migrating settings back (since I have to have most of my settings due to work deadlines) and things seemed better, but then suddenly the hangs started again.  I've been unable to isolate the culprit, presumed, setting... Unfortunately this is an extremely serious problem for me, but I'm not able to find a solution and Apple simply will not / cannot help.  

The one - possibly - common thread is that the hang predominantly occurs in Chrome proper ( and Epichrome instances) + when I copy or paste in such an app.  The hang DOES occur in other apps, but this seems to be closest combination to a reproduction I've been able to kind of isolate and yet this is not even 75% certain.  Obviously this kind of situation is the kiss of death for software engineering and I'm just trying to figure out options to rule out certain elements where possible.

At any rate, thanks for the news there!
